plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,347 @@
+			return err
+		}
+		headers["GWOS-API-TOKEN"] = g.authToken
+		statusCode, body, err = clients.SendRequest(http.MethodPost, g.Server+defaultMonitoringRoute, headers, nil, requestJSON)
+		if err != nil {
+			return err
+		}
+	}
+
+	if statusCode != 200 {
+		return fmt.Errorf("something went wrong during processing an http request[http_status = %d, body = %s]", statusCode, string(body))
+	}
+
+	return nil
+}
+
+func (g *Groundwork) Description() string {
+	return "Send telegraf metrics to GroundWork Monitor"
+}
+
+func init() {
+	outputs.Add("groundwork", func() telegraf.Output {
+		return &Groundwork{
+			ResourceTag:         "host",
+			DefaultHost:         "telegraf",
+			DefaultServiceState: string(transit.ServiceOk),
+		}
+	})
+}
+
+func (g *Groundwork) parseMetric(metric telegraf.Metric) (string, transit.DynamicMonitoredService) {
+	resource := g.DefaultHost
+	if value, present := metric.GetTag(g.ResourceTag); present {
+		resource = value
+	}
+
+	service := metric.Name()
+	if value, present := metric.GetTag("service"); present {
+		service = value
+	}
+
+	status := g.DefaultServiceState
+	value, statusPresent := metric.GetTag("status")
+	if validStatus(value) {
+		status = value
+	}
+
+	message := ""
+	if value, present := metric.GetTag("message"); present {
+		message = value
+	}
+
+	unitType := string(transit.UnitCounter)
+	if value, present := metric.GetTag("unitType"); present {
+		unitType = value
+	}
+
+	critical := -1.0
+	if value, present := metric.GetTag("critical"); present {
+		if s, err := strconv.ParseFloat(value, 64); err == nil {
+			critical = s
+		}
+	}
+
+	warning := -1.0
+	if value, present := metric.GetTag("warning"); present {
+		if s, err := strconv.ParseFloat(value, 64); err == nil {
+			warning = s
+		}
+	}
+
+	serviceObject := transit.DynamicMonitoredService{
+		BaseTransitData: transit.BaseTransitData{
+			Name:  service,
+			Type:  transit.Service,
+			Owner: resource,
+		},
+		Status:           transit.MonitorStatus(status),
+		LastCheckTime:    transit.MillisecondTimestamp{Time: metric.Time()},
+		LastPlugInOutput: message,
+		Metrics:          nil,
+	}
+
+	for _, value := range metric.FieldList() {
+		var thresholds []transit.ThresholdValue
+		thresholds = append(thresholds, transit.ThresholdValue{
+			SampleType: transit.Warning,
+			Label:      value.Key + "_wn",
+			Value: &transit.TypedValue{
+				ValueType:   transit.DoubleType,
+				DoubleValue: warning,
+			},
+		})
+		thresholds = append(thresholds, transit.ThresholdValue{
+			SampleType: transit.Critical,
+			Label:      value.Key + "_cr",
+			Value: &transit.TypedValue{
+				ValueType:   transit.DoubleType,
+				DoubleValue: critical,
+			},
+		})
+
+		valueType := transit.DoubleType
+		var floatVal float64
+		var stringVal string
+
+		switch value.Value.(type) {
+		case string:
+			valueType = transit.StringType
+			tmpStr := value.Value.(string)
+			stringVal = tmpStr
+		default:
+			floatVal, _ = internal.ToFloat64(value.Value)
```

Comment:
Error checking? It does not need to break the whole `parseMetric`, but just logging something like:
```go
g.Log.Warnf("Could not convert %s to float: %v", value.Key, err);
```

Reply:
Done